test/node-rpc-test.js

```diff
@@ -5,59 +5,780 @@
+    });
+  });
+
+  describe('getnetworkinfo', function () {
+    const check = (info, expected) => {
+      assert.ok(info);
+      assert.strictEqual(typeof info.version, 'string');
+      assert.strictEqual(typeof info.subversion, 'string');
+      assert.strictEqual(typeof info.protocolversion, 'number');
+      assert.strictEqual(typeof info.version, 'string');
+      assert.ok(isHex(info.localservices));
+      assert.strictEqual(info.localservices, expected.localservices);
+      assert.strictEqual(info.localrelay, expected.localrelay);
+      assert.strictEqual(info.timeoffset, 0);
+      assert.strictEqual(info.networkactive, true);
+      assert.strictEqual(info.connections, expected.connections);
+      assert.ok(Array.isArray(info.networks));
+      assert.strictEqual(info.networks.length, 0);
+      assert.strictEqual(typeof info.relayfee, 'number');
+      assert.strictEqual(info.incrementalfee, 0);
+      assert.ok(Array.isArray(info.localaddresses));
+      assert.strictEqual(info.localaddresses.length, expected.localaddrlen);
+
+      for (const addr of info.localaddresses) {
+        assert.strictEqual(typeof addr.address, 'string');
+        assert.strictEqual(typeof addr.port, 'number');
+        assert.strictEqual(typeof addr.score, 'number');
+      }
+
+      assert.strictEqual(info.warnings, '');
+    };
+
+    it('should get networkinfo (fullnode)', async () => {
+      const netinfo = await nclient.execute('getnetworkinfo');
+
+      check(netinfo, {
+        // NETWORK | WITNESS | BLOOM
+        localservices: '0000000d',
+        localrelay: true,
+        connections: 3,
+
+        // this node is listening.
+        localaddrlen: 1
+      });
+    });
+
+    it('should get networkinfo (pruned node)', async () => {
+      const netinfo = await prunednclient.execute('getnetworkinfo');
+
+      check(netinfo, {
+        // NETWORK | WITNESS
+        localservices: '00000009',
+        localrelay: true,
+        connections: 1,
+        localaddrlen: 0
+      });
+    });
+
+    it('should get networkinfo (spv node)', async () => {
+      const netinfo = await spvnclient.execute('getnetworkinfo');
+
+      check(netinfo, {
+        // WITNESS
+        localservices: '00000008',
+        localrelay: false,
+        connections: 1,
+        localaddrlen: 0
+      });
+    });
+  });
+
+  describe('gettxoutproof/verifytxoutproof', function() {
+    const TEST_WALLET = 'wallet-gettxoutproof';
+
+    const blockhashes = [];
+    const txids = [];
+
+    let mempoolTXID;
+
+    before(async () => {
+      // create wallet for testing.
+      await wclient.createWallet(TEST_WALLET);
+      await wclient.execute('selectwallet', [TEST_WALLET]);
+      const testAddr = await wclient.execute('getnewaddress');
+
+      // back to miner
+      await wclient.execute('selectwallet', [MINER_WALLET]);
+      const minerAddr = await wclient.execute('getnewaddress');
+
+      { // block 0 - This block will have 2 txs (including coinbase)
```

Comment:
These are generating blocks on the chain shared by all tests. Should the setup be a part of the before for all tests? It could potentially be generic enough. Currently this is the last test, so no other tests depend on it, however as more tests are added, they would always need to be added before the test to not depend state changes within these tests. Also, I don't think these tests require having all sets of different nodes, spv and etc.